src/Nethermind/Nethermind.Core/Specs/ForkActivation.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+
+namespace Nethermind.Core.Specs;
+
+public readonly struct ForkActivation : IEquatable<ForkActivation>, IComparable<ForkActivation>
+{
+    public long BlockNumber { get; }
+    public ulong Timestamp { get; }
+
+    public ForkActivation(long blockNumber, ulong timestamp = 0)
+    {
+        BlockNumber = blockNumber;
+        Timestamp = timestamp;
+    }
+
+    public static implicit operator ForkActivation(long blocknumber) => new(blocknumber);
+
+    public static implicit operator ForkActivation((long, ulong) blocknumberAndTimestamp)
+        => new(blocknumberAndTimestamp.Item1, blocknumberAndTimestamp.Item2);
+
+    public static implicit operator (long, ulong)(ForkActivation forkActivation)
```

Comment:
Maybe deconstruction would be better? https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/deconstruct#user-defined-types

Or do both.